bundles/org.openhab.binding.mybmw/src/main/java/org/openhab/binding/mybmw/internal/utils/Converter.java

```diff
@@ -12,89 +12,77 @@
  */
 package org.openhab.binding.mybmw.internal.utils;
 
-import java.lang.reflect.Type;
 import java.text.SimpleDateFormat;
-import java.time.LocalTime;
 import java.time.ZoneId;
-import java.time.ZoneOffset;
 import java.time.ZonedDateTime;
 import java.time.format.DateTimeFormatter;
-import java.util.ArrayList;
 import java.util.Date;
-import java.util.List;
 import java.util.Locale;
 import java.util.TimeZone;
 
 import org.eclipse.jdt.annotation.NonNullByDefault;
 import org.eclipse.jdt.annotation.Nullable;
-import org.openhab.binding.mybmw.internal.MyBMWConstants;
 import org.openhab.binding.mybmw.internal.dto.charge.Time;
-import org.openhab.binding.mybmw.internal.dto.properties.Address;
-import org.openhab.binding.mybmw.internal.dto.properties.Coordinates;
-import org.openhab.binding.mybmw.internal.dto.properties.Distance;
-import org.openhab.binding.mybmw.internal.dto.properties.Location;
-import org.openhab.binding.mybmw.internal.dto.properties.Range;
-import org.openhab.binding.mybmw.internal.dto.status.Mileage;
-import org.openhab.binding.mybmw.internal.dto.vehicle.Vehicle;
+import org.openhab.core.library.types.DateTimeType;
 import org.openhab.core.library.types.StringType;
 import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.google.gson.Gson;
-import com.google.gson.JsonArray;
-import com.google.gson.JsonObject;
-import com.google.gson.JsonParser;
-import com.google.gson.JsonSyntaxException;
-import com.google.gson.reflect.TypeToken;
-
 /**
  * The {@link Converter} Conversion Helpers
  *
  * @author Bernd Weymann - Initial contribution
+ * @author Martin Grassl - extract some methods to other classes
  */
 @NonNullByDefault
-public class Converter {
-    public static final Logger LOGGER = LoggerFactory.getLogger(Converter.class);
+public interface Converter {
+    static final Logger LOGGER = LoggerFactory.getLogger(Converter.class);
 
-    public static final String DATE_INPUT_PATTERN_STRING = "yyyy-MM-dd'T'HH:mm:ss";
-    public static final DateTimeFormatter DATE_INPUT_PATTERN = DateTimeFormatter.ofPattern(DATE_INPUT_PATTERN_STRING);
-    public static final DateTimeFormatter LOCALE_ENGLISH_TIMEFORMATTER = DateTimeFormatter.ofPattern("hh:mm a",
+    static final String DATE_INPUT_PATTERN_STRING = "yyyy-MM-dd'T'HH:mm:ss";
+    static final DateTimeFormatter DATE_INPUT_PATTERN = DateTimeFormatter.ofPattern(DATE_INPUT_PATTERN_STRING);
```

Comment:
```suggestion
    static final DateTimeFormatter DATE_INPUT_PATTERN = DateTimeFormatter.ISO_LOCAL_DATE_TIME;
```

Reply:
trying... getting errors like 
`java.time.format.DateTimeParseException: Text '2023-01-18T18:07:59.076Z' could not be parsed, unparsed text found at index 23` when using ISO_LOCAL_DATE_TIME and 
`java.time.format.DateTimeParseException: Text '2023-01-18T18:07:59.076Z' could not be parsed: Unable to obtain LocalDateTime from TemporalAccessor: {MilliOfSecond=76, InstantSeconds=1674065279, NanoOfSecond=76000000, MicroOfSecond=76000},ISO of type java.time.format.Parsed` when I use ISO_INSTANT. If none of the pre-defined formats work, then I'll stick to the pattern like used right now.

Reply:
This is only used for formatting, not for parsing, right? I see only one usage in line 56:
```java
format(Converter.DATE_INPUT_PATTERN)
```

Reply:
I completely reworked the time zone handling, looks like it is working now. The input can be parsed as Instant, not as LocalDateTime or ZonedDateTime...

Reply:
Yes, "2023-01-18T18:07:59.076Z" is UTC and can be directly parsed as an `Instant`. But that means it didn't work before the changes either? Anyway, glad you got it refactored and that it's working now.